tests/tracking/test_client.py

```diff
@@ -794,3 +795,14 @@ def test_client_log_metric_params_tags_overrides(mock_store):
     mock_store.log_batch_async.assert_called_once_with(
         run_id=run_id, metrics=metrics, params=params, tags=tags
     )
+
+
+def test_enable_async_logging(mock_store):
+    enable_async_logging(True)
+    MlflowClient().log_param(run_id="run_id", key="key", value="val")
+    mock_store.log_param_async.assert_called_once_with("run_id", Param("key", "val"))
+
+    MlflowClient().log_metric(run_id="run_id", key="key", value="val", step=1, timestamp=1)
+    mock_store.log_metric_async.assert_called_once_with("run_id", Metric("key", "val", 1, 1))
+    # Clean up the setup.
+    enable_async_logging(False)
```

Comment:
What would happen if we hit an error before `enable_async_logging(False)` is called? Would that affect other tests?